### PR TITLE
Generic dataset is added.

### PR DESCRIPTION
**Pull request by look4pritam**
Creates TFRecord files for training and validation for any generic dataset using following command.

python download_and_convert_data.py --dataset_name=generic --dataset_dir=/dataset/tensorflow/flowers/ --source_dir=/dataset/flowers/

Here in this example
--source_dir contains flowers dataset but any other dataset can be input.

Then train for given dataset using following command.

python train_image_classifier.py --train_dir=/workspace/models/generic --dataset_name=generic --dataset_split_name=train --dataset_dir=/dataset/tensorflow/flowers --model_name=inception_v3 --save_interval_secs=600 --save_summaries_secs=120 --log_every_n_steps=50 --max_number_of_steps=200

Finally evaluate using following command.

python eval_image_classifier.py --checkpoint_path=/workspace/models/generic --eval_dir=/workspace/models/generic --dataset_name=generic --dataset_split_name=validation --dataset_dir=/dataset/tensorflow/flowers --model_name=inception_v3

### CODE REVIEW

**Comment by look4pritam** [comment]
Details of committed files.

1)
research/slim/datasets/convert_generic.py - It is similar to research/slim/datasets/download_and_convert_flowers.py

2)
research/slim/datasets/dataset_factory.py - generic dataset is added.

3)
research/slim/datasets/dataset_utils.py - metadata file read and write is added.
metadata file contains - number of classes, number of validation images, number of training images as metadata

4)
research/slim/datasets/generic.py - It is similar to research/slim/datasets/flowers.py

5)
research/slim/download_and_convert_data.py - generic dataset is added.
source_dir - dataset source directory is added as command line argument

6)
research/slim/nets/nets_factory.py - model_name_map is added, which is used at inference time

7)
research/slim/train_image_classifier.py - training accuracy is added

**Comment by look4pritam** [comment]
I have utilized this contribution to train following models
- inception_v3
- inception_resnet_v2


I have utilized MS-Celeb-1M dataset.
- 20K persons dataset
- 90% images - training
- 10% images - validation

I have achieved more than 99% percent validation accuracy.

Currently I am working for MS-Celeb-1M dataset
- 100K persons dataset
- 90% images - training
- 10% images - validation

File: research/slim/nets/nets_factory.py [hunk 1]
```diff
@@ -93,6 +93,34 @@
...
+                  'resnet_v1_200': 'resnet_v1_200',
+                  'resnet_v2_50': 'resnet_v2_50',
+                  'resnet_v2_101': 'resnet_v2_101',
+                  'resnet_v2_152': 'resnet_v2_152',
+                  'resnet_v2_200': 'resnet_v2_200',
+                  'mobilenet_v1': 'MobilenetV1',
+                  'mobilenet_v1_075': 'MobilenetV1',
+                  'mobilenet_v1_050': 'MobilenetV1',
+                  'mobilenet_v1_025': 'MobilenetV1',
+                  #'nasnet_cifar': 'nasnet_cifar',
```
Comment (jkholodnov) [inline research/slim/nets/nets_factory.py]: delete commented

Comment (look4pritam) [reply research/slim/nets/nets_factory.py]: I have added it to complete code. If you sure of it, I will remove the comment.

**Comment by look4pritam** [comment]
Is anything else is needed?
